llpc/tool/amdllpc.cpp

```diff
@@ -341,6 +341,42 @@ struct CompileInfo
                                                                 // same as specified pipeline layout
 };
 
+// =====================================================================================================================
+// Checks whether the input data is actually a ELF binary
+static bool IsElfBinary(
+    const void* pData,    // [in] Input data to check
+    size_t      dataSize) // Size of the input data
+{
+    bool isElfBin = false;
+    if (dataSize >= sizeof(Elf64::FormatHeader))
+    {
+        auto pHeader = reinterpret_cast<const Elf64::FormatHeader*>(pData);
```

Comment:
nit: `auto* pHeader = ...`

Reply:
Again, that's orthogonal to my change, which just moves this code from one place to another.